starlette/datastructures.py

```diff
@@ -290,6 +291,95 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}({repr(items)})"
 
 
+class MultiDict(ImmutableMultiDict):
+    def __setitem__(self, key: typing.Any, value: typing.Any) -> None:
+        self.setlist(key, [value])
+        self._dict[key] = value
+
+    def __delitem__(self, key: typing.Any) -> None:
+        self._list = list(filter(lambda item: item[0] != key, self._list))
+        del self._dict[key]
+
+    def pop(self, key: typing.Any) -> typing.Any:
+        list_ = list(filter(lambda item: item[0] == key, self._list))
+        poped = self._dict.pop(key)
+        self.setlist(key, [v for _, v in list_[:-1]])
+        return poped
```

Comment:
I think the behavior of the standard `Mapping` operations should match `dict`.
So, `pop` should remove the item entirely. (Rather than remove one value from the list)

Also we're missing the `default` argument.

```python
def pop(self, key: typing.Any, default: typing.Any=None) -> typing.Any:
    self._list = [(existing_key, value) for (existing_key, value) in self._list if existing_key != key]
    return self._dict.pop(key, default)
```

Or...

```python
def pop(self, key: typing.Any, default: typing.Any=None) -> typing.Any:
    self._list = list(filter(lambda item: item[0] != key, self._list))
    return self._dict.pop(key, default)
```